### ISSUE TITLE
ScrollArea::show_rows() does not scroll smoothly with very large total_rows

### ISSUE THREAD

**Issue by parasyte** [emilk/egui#1391]
**Describe the bug**

I have a UI with about 16 million rows (but this is on the low end of what I want to support). I'm using `ScrollArea::show_rows()` to keep a sane viewport size.

Up to about 2 million rows, scrolling with the trackpad or by dragging the background with a mouse is relatively smooth. Above 2 million total rows, scrolling has some obvious hitching. Above 100 million total rows, the scrolling gets very broken.

**To Reproduce**
1. Patch the demo lib to increase the `num_rows` to `16_000_000`: https://github.com/emilk/egui/blob/ccbddcfe951e01c55efd0ed19f2f2ab5edfad5d9/egui_demo_lib/src/apps/demo/scrolling.rs#L85
2. Run the demo and go to the "Scolling" demo and then the "Scroll a lot of lines" tab.
3. Scroll with the trackpad or by dragging the background with the mouse.
4. You'll see it does not scroll smoothly.
5. Increase `num_rows` to `160_000_000` and try the same test.
6. Scrolling is almost completely unusable with this many rows.

**Expected behavior**

Always scroll smoothly regardless of the total number of rows.

**Desktop (please complete the following information):**
 - OS: Windows 11, macOS 12.2.1

**Comment by emilk** [comment]
Yup, `f32` will cause trouble here. Seems like we need to switch to `f64` in `ScrollArea::show_rows`, which should be pretty straight-forward.
